File: DotNet/Framework/AutomationFramework/Helpers/Drivers/DriverFactory.cs
using Helpers.Logging;

namespace Helpers.Drivers
{
    public class DriverFactory
    {
        public IDriver driver;
        private DriverType driverType;

        public DriverFactory(DriverType type)
        {
            driverType = type;
        }

        public IDriver GetDriverType()
        {
            driver = driverType switch
            {
                DriverType.Chrome => new Chrome(),
                DriverType.Edge => new Edge(),
                DriverType.Firefox => throw new NotImplementedException(),
                _ => throw new Exception("Enter the valid browser")
            };
         return driver;
        }
    }
}

File: Java/AutomationFramework/src/com/app/helpers/drivers/DriverFactory.java
package com.app.helpers.drivers;

public class DriverFactory {
    IDriver driver;
    DriverType driverType;

    public DriverFactory(DriverType type) {
        driverType = type;
    }

    public IDriver GetDriverType() throws Exception {
        driver = switch(driverType) {
            case Chrome -> new Chrome();
            case Edge -> new Edge();
            default -> throw new Exception("Please provide valid browser");
        };
        return driver;
    }
}
